mlserver/parallel/registry.py

```diff
@@ -76,18 +75,37 @@ async def _handle_worker_stop(self, signum, frame):
         )
 
     async def _get_or_create(self, model: MLModel) -> InferencePool:
-        env_tarball = _get_env_tarball(model)
-        if not env_tarball:
-            return self._default_pool
-
-        env_hash = await compute_hash(env_tarball)
-        if env_hash in self._pools:
-            return self._pools[env_hash]
-
-        env = await self._extract_tarball(env_hash, env_tarball)
-        pool = InferencePool(
-            self._settings, env=env, on_worker_stop=self._on_worker_stop
-        )
+        if model.settings.parameters and model.settings.parameters.environment_path:
+            environment_path = os.path.abspath(
+                os.path.expanduser(
+                    os.path.expandvars(model.settings.parameters.environment_path)
+                )
+            )
+            logger.info(f"Using environment {environment_path}")
+            env_hash = await compute_hash_of_string(environment_path)
+            if env_hash in self._pools:
+                return self._pools[env_hash]
+            env = Environment(
+                env_path=environment_path,
+                env_hash=env_hash,
+                delete_env=False,
+            )
```

Comment:
perhaps extract in a specific function to make it more legible and also add unit tests for it?

Reply:
I've refactored it into two methods `_get_or_create_with_existing_env` and `_get_or_create_with_tarball`